Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -1753,10 +1753,15 @@ Normalize(std::vector<double> & x)
 static bool
 IsAffine(const mat44 & nifti_mat)
 {
-  const vnl_matrix_fixed<float, 4, 4> mat(&(nifti_mat.m[0][0]));
+  vnl_matrix_fixed<double, 4, 4> mat;
+
+  for (int i = 0; i < 4; ++i)
+    for (int j = 0; j < 4; ++j)
+      mat[i][j] = double{ nifti_mat.m[i][j] };
+
   // First make sure the bottom row meets the condition that it is (0, 0, 0, 1)
   {
-    float bottom_row_error = itk::Math::abs(mat[3][3] - 1.0f);
+    double bottom_row_error = itk::Math::abs(mat[3][3] - 1.0);
```

Comment:
is _float_ epsilon several lines below (line 1769) intentional
```cpp
if (bottom_row_error > std::numeric_limits<float>::epsilon())
```
I thought that it might have been forgotten.

Reply:
> is _float_ epsilon several lines below (line 1769) intentional?

@issakomi Good question. But I wonder, isn't `std::numeric_limits<T>::epsilon()` rather arbitrary anyway? I just don't know, but if `mat[3][i]` must be close to zero (for `i` = 0, 1, 2), then those matrix elements might as well be compared against `std::numeric_limits<T>::min()`, which if much smaller than `std::numeric_limits<T>::epsilon()`.

Honestly I'm just wondering what is intended as well. I have no suggestion here.

Reply:
I think the bottom row should be precisely (0,0,0,1) - but I guess it's possible there might be some epsilon error when the inverse is calculated

Reply:
I think it was just forgotten.

Reply:
> there might be some epsilon error when the inverse is calculated

OK, but `std::numeric_limits<T>::epsilon()` isn't just _some_ epsilon error. It is specifically the difference between 1 and the least value greater than 1 that is representable by floating point type `T`. https://www.cplusplus.com/reference/cfloat/ That's why I wonder if this specific epsilon should also be used for values that should be close to zero (rather than close to one). But I just don't know, honestly.

Reply:
One further idea here, currently the test is made with the sum as each entry added. Should we instead test the individual entries?

Changing to `::min()`, or testing the individual entries will be substantially stricter than the existing implementation as well.